game/src/main/java/org/apollo/game/io/player/JdbcPlayerSerializer.java

```diff
@@ -1,33 +1,464 @@
+			GET_ITEMS_QUERY = "SELECT inventory_id, slot, item_id, quantity FROM get_items(?::text)",
+			GET_ATTRIBUTES_QUERY = "SELECT attr_type, name, value FROM get_attributes(?::text);",
+			GET_STATS_QUERY = "SELECT skill, stat, experience FROM get_skills(?::text)",
+			SET_ACCOUNT_QUERY = "CALL set_account(?::citext, ?::rank)",
+			SET_PLAYER_QUERY = "CALL set_player(?::citext, ?::text, ?, ?, ?, ?)",
+			SET_APPEARANCE_QUERY = "CALL set_appearance(?::text, ?::gender, ?, ?)",
+			SET_STAT_QUERY = "CALL set_stat(?::skill, ?, ?, ?::text)",
+			SET_ATTRIBUTE_QUERY = "CALL set_attribute(?::text, ?::attribute_type, ?::varchar, ?::text)",
+			SET_ITEM_QUERY = "CALL set_item(?::text, ?, ?, ?, ?)",
+			DELETE_ITEM_QUERY = "CALL delete_item(?::text, ?, ?)";
 
 	/**
-	 * Creates the JdbcPlayerSerializer.
+	 * The config id's of known inventories.
+	 */
+	private static final int ITEM_BAG_ID = 93,
+			WORN_EQUIPMENT_ID = 94,
+			BANK_ID = 95;
+
+	/**
+	 * A factory method to construct a new {@link JdbcPlayerSerializer}.
+	 *
+	 * @param world       The game world to feed to {@link Player} instances.
+	 * @param connections The supplier of database connections.
+	 * @return The {@link JdbcPlayerSerializer}.
+	 */
+	public static JdbcPlayerSerializer create(World world, ConnectionSupplier connections) {
+		return new JdbcPlayerSerializer(world, connections);
+	}
+
+	/**
+	 * Supplies this serializer with database connections.
+	 */
+	private final ConnectionSupplier connections;
+
+	/**
+	 * Creates the {@link JdbcPlayerSerializer}.
 	 *
-	 * @param world The {@link World} to place the {@link Player}s in.
+	 * @param world       The {@link World} to place the {@link Player}s in.
+	 * @param connections The supplier of database connections.
 	 */
-	public JdbcPlayerSerializer(World world) {
+	private JdbcPlayerSerializer(World world, ConnectionSupplier connections) {
 		super(world);
+
+		this.connections = connections;
 	}
 
 	@Override
 	public void savePlayer(Player player) throws Exception {
-		throw new UnsupportedOperationException("JDBC saving is not supported at this time.");
+		try (Connection connection = connections.get()) {
+			Email email = Email.of(player.getCredentials().getUsername());
+			String displayName = player.getCredentials().getUsername();
+
+			putRank(connection, email, player.getPrivilegeLevel());
+			putPlayer(connection, email, player);
+			putAppearance(connection, displayName, player.getAppearance());
+			putStats(connection, displayName, player.getSkillSet());
+
+			putInventory(connection, displayName, ITEM_BAG_ID, player.getInventory());
+			putInventory(connection, displayName, WORN_EQUIPMENT_ID, player.getEquipment());
+			putInventory(connection, displayName, BANK_ID, player.getBank());
+
+			putAttributes(connection, displayName, player.getAttributes());
+		}
+	}
+
+	private void putRank(Connection connection, Email email, PrivilegeLevel rank) throws SQLException {
```

Comment:
I don't like the amount of manual interaction with `PreparedStatement`s and data mapping we have to do in here. I'd prefer if we used something like Spring's `JDBCTemplate` or jOOQ to handle splitting out the data mapping code (into spring `RowMapper`s or jOOQ `Converters`), common JDBC errors, and replace positional parameters with named parameters.